src/main/scala/com/gossiperl/client/GossiperlProxy.scala

```diff
@@ -0,0 +1,131 @@
+package com.gossiperl.client
+
+import akka.actor.{ActorRef, ActorLogging, Actor}
+import akka.util.Timeout
+import com.gossiperl.client.SupervisorProtocol.SupervisorAction
+import com.gossiperl.client.actors.ActorRegistry
+import com.gossiperl.client.serialization.{Serializer, DeserializeResult, CustomDigestField}
+
+import scala.concurrent.{Promise, Future}
+import scala.util.{Try, Failure, Success}
+
+import concurrent.duration._
+
+object GossiperlProxyProtocol {
+  case class Connected( configuration: OverlayConfiguration )
+  case class Disconnected( configuration: OverlayConfiguration )
+  case class Event( configuration: OverlayConfiguration, eventType: String, member: Any, heartbeat: Long )
+  case class SubscribeAck( configuration: OverlayConfiguration, eventTypes: Seq[String] )
+  case class UnsubscribeAck( configuration: OverlayConfiguration, eventTypes: Seq[String] )
+  case class ForwardAck( configuration: OverlayConfiguration, replyId: String )
+  case class Forward( configuration: OverlayConfiguration, digestType: String, binaryEnvelope: Array[Byte], envelopeId: String )
+  case class Error( configuration: OverlayConfiguration, error: Throwable )
+
+  case class ShutdownRequest( p: Promise[ActorRef] )
+}
+
+class GossiperlProxy( val configuration: OverlayConfiguration, p: Promise[GossiperlProxy] ) extends ActorRegistry with ActorLogging {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+  implicit val timeout = Timeout(1 seconds)
+
+  // Yes, we do want it like this. We want the proxy to be available as an instance to the caller so we can simply forward events to it.
+  // Proxy is used from outside of the actor system.
+  p.success(this)
```

Comment:
An actor instance should never be available to an outside component. Everything should talk to an actor by using its actor reference. This concept becomes even more important when you start using routers.